### ISSUE TITLE
Better Module/Disable and Install behavior

### ISSUE THREAD

**Issue by buddh4** [humhub/humhub#3737]
Currently disabling a module completely removes the modules tables and data and removes the module from `module_enabled`.

It should be possible to disable a module without flushing the data. The following issues need to be considered:

-  Search
- Content (Stream)
- Integrity
- Events

If a module update or installation (migration) fails, the module should be automatically set to disabled until the issues are resolved.

Ideally, there should be a migration log category, in order to troubleshoot the issues.

**Comment by starbock** [comment]
This one is pretty important, so glad to see it already being addressed.  For example, some regulatory compliance issue emerges that REAQUIRES taking some functionality off-line (e.g., Files module) while remediation steps are taken, then the module can be brought back on-line.  Deleting all content prevents this process.  Another example could be an upgrade breaking some module functionality, requiring it to be taken off-line while a fix is implemented;  without the ability to disable-without-delete it is operationally difficult to manage through this temporary condition.  There are lots of real-world cases like that, thus appreciate consideration to implement this capability.

**Comment by martin-rueegg** [comment]
We too see this as a requirement and have so fare implemented it in our module internally. But it would certainly be better suited - and handled - by the core.

In fact I would see the current "installed-but-not-activated" state as something like this. And that only "uninstalling" the module would actually delete the content.

Happy to contribute to this.

@luke- ?

**Comment by luke-** [comment]
The main problem here is that a module can have many records like notifications / stream entries / activities in the database. We somehow find a way to skip these records when a module is "paused".

 As with @martin-rueegg  discussed earlier a `class_map` table could be also helpful here, the table could have the following attributes `id`, `class_name`, `active`.

When a module is paused, all classes with the namespace are changed to `active=0`. On stream/activity/notification queries we can easily filter out non-active classes this way.

**Comment by luke-** [comment]
Yes, `module_id` would really be better here, even if a join is more...
